svcdef/resolvetypes.go

```diff
@@ -0,0 +1,71 @@
+package svcdef
+
+// typeBox is used where we know that either a Message or an Enum will need to
+// be returned, but not both, and not neither. If one field is nil, the other
+// is expected to be a valid pointer.
+type typeBox struct {
+	Message *Message
+	Enum    *Enum
+}
+
+// resolveTypes accepts a pointer to a Svcdef and modifies that Svcdef, and its
+// child structs, in place. After the initial creation of Svcdef, each
+// FieldType will have it's name set to the name of the type that it
+// represents, but it won't have a pointer to the full instance of the type
+// with that name. Type resolution is the process of taking the name of each
+// FieldType, then searching for the enum/message with that name and setting
+// the respective *Message/*Enum property of our FieldType to point to that
+// enum/message.
+func resolveTypes(sd *Svcdef) {
+	tmap := newTypeMap(sd)
+	for _, m := range sd.Messages {
```

Comment:
should `sd.Enums` be iterated over here? Likely no, I believe Enums still have to be used inside a Message and are exposed through `m.Fields`, but this is not obvious if you don't know protobuf or simply might not be in mind. Consider adding a short comment to remind the reader.